### PR TITLE
Add ConnectorClientOptions to BotFrameworkAdapterSettings 

### LINKED ISSUE #2013: Add ConnectorClientOptions to BotFrameworkAdapterSettings
**Describe the solution you'd like**
The ability to set `ServiceClientOptions` when ConnectorClients are created

**Describe alternatives you've considered**
Hacking the `createConnectorClientInternal` method prototype on `BotFrameworkAdapter`, and accessing the ConnectorClient's properties directly:
```js
BotFrameworkAdapter.prototype.createConnectorClientInternal = function (serviceUrl, credentials) {
    // Set noRetryPolicy: true so ServiceClient code does not add one.
    // This code adds the default retry policies, with a modified exponentialRetryPolicy
    const client = new ConnectorClient(credentials, { baseUri: serviceUrl, noRetryPolicy: true });
    
    //retryCount, retryInterval, minRetryInterval, maxRetryInterval, requestTimeout
    client._requestPolicyFactories.push(retryWithTimeoutPolicy(3, 1000 * 2, 1000 * 1, 1000 * 4, 1000 * 5));
    client._requestPolicyFactories.push(systemErrorRetryPolicy());
    client._requestPolicyFactories.push(throttlingRetryPolicy());

    return client;
};
```

[enhancement]

### PR DESCRIPTION
**Pull request by EricDahlvang**
Fixes #2013

### CODE REVIEW

File: libraries/botbuilder/src/botFrameworkAdapter.ts [hunk 1]
```diff
@@ -8,9 +8,10 @@
 
 import { STATUS_CODES } from 'http';
 import { arch, release, type } from 'os';
+import { HttpClient } from '@azure/ms-rest-js';
```
Comment (stevengum) [inline libraries/botbuilder/src/botFrameworkAdapter.ts:11]: Is this the interface or an abstract class?

Comment (EricDahlvang) [reply libraries/botbuilder/src/botFrameworkAdapter.ts:11]: It is just the interface. I wondered if i should leave it out.?

Comment (stevengum) [reply libraries/botbuilder/src/botFrameworkAdapter.ts:11]: If it’s only used in the private method as a parameter I would leave it out, change the type to any and perhaps leave a comment.

Comment (EricDahlvang) [reply libraries/botbuilder/src/botFrameworkAdapter.ts]: removed

**Pull request merged**